lib/proxy/rpmsg_retarget.c

```diff
@@ -228,7 +228,7 @@ int _read(int fd, char *buffer, int buflen)
 	syscall.args.int_field2 = buflen;
 	syscall.args.data_len = 0;	/*not used */
 
-	resp = (struct rpmsg_rpc_syscall *)tmpbuf;
+	resp = (struct rpmsg_rpc_syscall *)(void *)tmpbuf;
```

Comment:
is `resp = (void *)tmpbuf;`  ok?
if yes few optimizations are possible in your PR

Reply:
Yes, it is OK. I will simplify these bits. It should make the code easier to read.